tool/dumper/vkgcPipelineDumper.cpp

```diff
@@ -1017,7 +1017,8 @@ void PipelineDumper::updateHashForNonFragmentState(const GraphicsPipelineBuildIn
   auto iaState = &pipeline->iaState;
   if (enableNgg)
     hasher->Update(iaState->topology);
-  hasher->Update(iaState->patchControlPoints);
+  if (pipeline->gs.pModuleData || pipeline->tcs.pModuleData || pipeline->tes.pModuleData)
+    hasher->Update(iaState->patchControlPoints);
```

Comment:
Note that if these shader stages exist in the pipeline, then I believe patchControlPoints can effect the code for the vertex shader as well by changing the way that the outputs of the vertex shaders are handled.

This is why we do not just checking if we are compiling a relocatable vertex shader.